yt/sample_data/api.py

```diff
@@ -132,7 +132,15 @@ def get_data_registry_table():
 def _get_test_data_dir_path():
     from yt.config import ytcfg
 
-    return Path(ytcfg.get("yt", "test_data_dir"))
+    p = Path(ytcfg.get("yt", "test_data_dir"))
+    if p.is_dir():
+        return p
+    warn(
+        "Storage directory from yt config doesn't exist "
```

Comment:
This is an improvement.  I still think that I have a slight preference for allowing pooch to set this, but I am not going to push it, as this may genuinely be better for the use cases commonly found.

Reply:
I've inspected pooch's source code and as far as I understand, it _requires_ that the storage dir be specified explicitly.
We could propose a patch upstream though, but it's not feasible in this PR, unless I'm mistaken.